ports/stm32/accel.c

```diff
@@ -185,22 +259,60 @@ STATIC mp_obj_t pyb_accel_filtered_xyz(mp_obj_t self_in) {
         }
         tuple[i] = mp_obj_new_int(val);
     }
+#elif MICROPY_HW_HAS_KXTJ3
+    uint8_t data[5];
+    int32_t data_sum[NUM_AXIS];
+
+    int i=0;
+    for ( ; i<NUM_AXIS; i++) {
+        data_sum[i] = 0;
+    }
 
+    int j=0;
+    while(1) {
+        data[0] = KXT_REG_XOUT_H;
+        i2c_writeto(I2C1, KXT_ADDR, data, 1, false);
+        i2c_readfrom(I2C1, KXT_ADDR, data, 5, true);
+        for (i = 0; i < NUM_AXIS; i++) {
+            data_sum[i] += (int8_t)data[i*2];
+        }
+        // no need to waste time for the last loop 
+        if (j >= FILT_DEPTH - 1) {
+            break;
+        }
+        // ODR is 200Hz with 10ms LP filter roll-over
+        mp_hal_delay_ms(10+1);
+        j++;
+    }
```

Comment:
This filtering routine accumulates all the FILT_DEPTH samples each time the method is called.  But for the MMA it just takes one sample per call and stores the results in the rolling buffer "buf".  Regardless of whether one approach is better or not, it should be that the KXT uses the same algorithm as the MMA.  It should be simple to do this, just reuse the MMA code and select with a macro how the data is read in for the filtering.

Reply:
Yep, you're right. I did the changes in the code. Only one access to the component per method call.